server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java

```diff
@@ -70,6 +73,17 @@ public FileInfo(String name, StoreFileMetadata metadata, ByteSizeValue partSize)
             assert IntStream.range(0, numberOfParts).mapToLong(this::partBytes).sum() == metadata.length();
         }
 
+        public FileInfo(StreamInput in) throws IOException {
+            this(in.readString(), new StoreFileMetadata(in), new ByteSizeValue(in));
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(name);
+            metadata.writeTo(out);
+            partSize.writeTo(out);
```

Comment:
I am not entirely sure it is true, but looking at the constructor and `toXContent`, it looks like `partSize` can be null. Can we either allow null in streaming or tighten the constructor to not all null.

Reply:
It seems like `BlobStoreRepository#chunkSize` allows it to be null, even though none of the current implementations return null. I'll allow null values just to be safe.